### PR TITLE
feat(bigquery/storage/managedwriter): expose connection multiplexing as experimental

### PR DESCRIPTION
**Pull request by shollyman**
This PR exposes the necessary options to control the new experimental multiplexing features within the managedwriter package, and attempts to document them sufficiently for correct consumption.

Towards: https://github.com/googleapis/google-cloud-go/issues/7103

### CODE REVIEW

**Review by yirutang — APPROVED** [review]
Description looks good to me.

File: bigquery/storage/managedwriter/options.go [hunk 1]
```diff
@@ -48,31 +61,55 @@ type writerClientOption interface {
...
+// be in flight on a connection at a time.  This limit is enforced on all connections
+// created by the instantiated Client.
 //
-// TODO: export this as part of the multiplex feature launch.
-func defaultMaxInflightRequests(n int) option.ClientOption {
+// Note: the WithMaxInflightRequests WriterOption can still be used to control
+// the behavior for individual ManagedStream writers when not using multiplexing.
+//
+// This ClientOption is EXPERIMENTAL and subject to change.
+func DefaultMaxInflightRequests(n int) option.ClientOption {
```
Comment (codyoss) [inline bigquery/storage/managedwriter/options.go:112]: nit: most of our options are prefixed with With, could we carry that convention here as well?

File: bigquery/storage/managedwriter/options.go [hunk 2]
```diff
@@ -61,14 +61,14 @@ type writerClientOption interface {
...
-// EnableMultiplexing is an EXPERIMENTAL option that enables connection sharing
+// WithMultiplexing is an EXPERIMENTAL option that controls connection sharing
 // when instantiating the Client.  Only writes to default streams can leverage the
 // multiplex pool.  Internally, the client maintains a pool of connections per BigQuery
 // destination region, and will grow the pool to it's maximum allowed size if there's
 // sufficient traffic on the shared connection(s).
 //
 // This ClientOption is EXPERIMENTAL and subject to change.
-func EnableMultiplexing(enable bool) option.ClientOption {
+func WithMultiplexing(enable bool) option.ClientOption {
```
Comment (codyoss) [inline bigquery/storage/managedwriter/options.go:71]: This one I think is fine to not use the With and instead pass nothing. The presence of the option seems like a logical true if passed. But I will leave it up to you

**Review by codyoss — APPROVED** [review]
Left one more consideration, but I am fine either way

File: bigquery/storage/managedwriter/options.go (hunk 2 above)
Comment (shollyman) [reply bigquery/storage/managedwriter/options.go]: Decided to do `WithMultiplexing()` here without argument(s).

**Pull request merged**